run.sh

```diff
@@ -1,22 +1,46 @@
+#!/bin/bash
+
 # note: must be run from the project root folder
 
+# set env variables
+envFile="$PWD/.env"
+source "$envFile"
+BOOTSTRAP_SERVER=$BOOTSTRAP_SERVER
+DEPOSIT_GROUP=$DEPOSIT_GROUP
+DEPOSIT_TOPIC=$DEPOSIT_TOPIC
+DEPOSIT_BUCKET_NAME=$DEPOSIT_BUCKET_NAME
+DEPOSIT_KEY_NAME=$DEPOSIT_KEY_NAME
+K_AWS_ACCESS_KEY_ID=$K_AWS_ACCESS_KEY_ID
+K_AWS_SECRET_ACCESS_KEY=$K_AWS_SECRET_ACCESS_KEY
+K_AWS_SESSION_TOKEN=$K_AWS_SESSION_TOKEN
+K_AWS_EXPIRATION=$K_AWS_EXPIRATION
+HEADER_ACCEPT=$HEADER_ACCEPT
+HEADER_X_API_KEY=$HEADER_X_API_KEY
+API_ENDPOINT=$API_ENDPOINT
+KAFKA_TYPE=$KAFKA_TYPE
+CONFLUENT_KEY=$CONFLUENT_KEY
+CONFLUENT_SECRET=$CONFLUENT_SECRET
+
 # build
 echo "Compiling."
 mvn clean package assembly:single
 
 # run
 echo "Executing."
 java -Dlogback.configurationFile=./src/main/resources/logback.xml \
-	-jar ./target/jpo-aws-depositor-jar-with-dependencies.jar \
-	--bootstrap-server $DOCKER_HOST_IP:9092 \
-	-g group1 \
-	-t $DEPOSIT_TOPIC \
-	-b $DEPOSIT_BUCKET_NAME \
-	-k $DEPOSIT_KEY_NAME \
-	-i $K_AWS_ACCESS_KEY_ID \
-	-a $K_AWS_SECRET_ACCESS_SECRET \
-	-n $K_AWS_SESSION_TOKEN \
-	-e $K_AWS_EXPIRATION \
-	-u $API_ENDPOINT \
-	-h $HEADER_ACCEPT \
-	-x $HEADER_X_API_KEY
\ No newline at end of file
+	-DBOOTSTRAP_SERVER=$BOOTSTRAP_SERVER \
+	-DDEPOSIT_GROUP=$DEPOSIT_GROUP \
+	-DDEPOSIT_TOPIC=$DEPOSIT_TOPIC \
+	-DDEPOSIT_BUCKET_NAME=$DEPOSIT_BUCKET_NAME \
+	-DDEPOSIT_KEY_NAME=$DEPOSIT_KEY_NAME \
+	-DK_AWS_ACCESS_KEY_ID=$K_AWS_ACCESS_KEY_ID \
+	-DK_AWS_SECRET_ACCESS_KEY=$K_AWS_SECRET_ACCESS_KEY \
+	-DK_AWS_SESSION_TOKEN=$K_AWS_SESSION_TOKEN \
+	-DK_AWS_EXPIRATION=$K_AWS_EXPIRATION \
+	-DHEADER_ACCEPT=$HEADER_ACCEPT \
+	-DHEADER_X_API_KEY=$HEADER_X_API_KEY \
+	-DAPI_ENDPOINT=$API_ENDPOINT \
+	-DKAFKA_TYPE=$KAFKA_TYPE \
+	-DCONFLUENT_KEY=$CONFLUENT_KEY \
+	-DCONFLUENT_SECRET=$CONFLUENT_SECRET \
+	-jar ./target/jpo-aws-depositor-jar-with-dependencies.jar
```

Comment:
Is this necessary? I would think that the JAR would have access to the environment variables without them being passed in directly.

Reply:
These have been removed. The updated code accesses environment variables instead of the values passed in as system properties as seen here.